tests/tools/onert_train/src/metric.cc

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) 2023 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "metric.h"
+#include "nnfw_util.h"
+
+#include <algorithm>
+#include <stdexcept>
+
+namespace onert_train
+{
+
+Metric::Metric(const std::vector<Allocation> &output, const std::vector<Allocation> &expected,
+               const std::vector<nnfw_tensorinfo> &infos)
+  : _output{output}, _expected{expected}, _infos{infos}
+{
+  // DO NOTHING
+}
+
+template <typename T>
+float Metric::categoricalAccuracy(const T *output, const T *expected, int32_t batch, int32_t size)
```

Comment:
@jyoungyun May I ask why we use `int32_t` for `size`?
- signed type halves the max size.
- Also, we need to check whether it is negative. (Currently, it does not check.)

Reply:
I fixed it to use `uint64_t` for `size`.